app/Models/User.php

```diff
@@ -235,4 +237,17 @@ public function getMastodonUrlAttribute(): ?string {
     public function notifications(): MorphMany {
         return $this->morphMany(DatabaseNotification::class, 'notifiable')->orderBy('created_at', 'desc');
     }
+
+    /**
+     * @throws RateLimitExceededException
+     */
+    public function sendEmailVerificationNotification(): void {
+        $mailLimiter = Cache::get('verification-mail-sent-' . $this->id);
+        if (isset($mailLimiter)) {
+            throw new RateLimitExceededException();
+        }
+
+        Cache::put('verification-mail-sent-' . $this->id, time(), now()->addMinutes(5));
+        parent::sendEmailVerificationNotification();
```

Comment:
Ich würde gerne den [Laravel RateLimiter](https://laravel.com/docs/9.x/rate-limiting#basic-usage) nutzen. Der ist eigentlich ganz ähnlich zu dem, was du gebaut hast, beschreibt aber IMO besser, was er tut:

```suggestion
        Log::info("Attempting to send verification email for user#" . $this->id);

        $executed = RateLimiter::attempt(
            'verification-mail-sent-' . $this->id,
            $maxAttempts = 1,
            function() {
                parent::sendEmailVerificationNotification();
                Log::info("Sent the verification email for user#" . $this->id . " successfully.");
            },
            $decaySeconds = 5 * 60
        );

        if (!$executed) {
            Log::info("Sending the verification email for user#" . $this->id . " was rate-limited.");
            throw new RateLimitExceededException();
        }
```

Oh, und weil wir ja jetzt mehr loggen wollen, notiert er sich die Sachen auch noch ins Log rein.